StepLang/Framework/Conversion/LengthFunction.cs

```diff
@@ -0,0 +1,49 @@
+using StepLang.Interpreting;
+using StepLang.Parsing.Expressions;
+
+namespace StepLang.Framework.Conversion;
+
+public class LengthFunction : NativeFunction
+{
+    public const string Identifier = "length";
+
+    public override async Task<ExpressionResult> EvaluateAsync(Interpreter interpreter,
+        IReadOnlyList<Expression> arguments,
+        CancellationToken cancellationToken = default)
+    {
+        if (arguments.Count is not 1)
+            throw new InvalidArgumentCountException(1, arguments.Count);
+
+        var exp = arguments.Single();
+
+        int length;
+
+        switch (exp)
+        {
+            case ConstantExpression conExp:
+                var str = conExp.Result.ExpectString();
```

Comment:
this wont work for other constant expressions as `ExpectString` also checks the type (`ResultType`) of the expression.

Either:
- evaluate always before doing something with the result, or
- add another switch for the result type in this branch

Reply:
You always need to be able to handle `VariableExpressions`, so I'd recommend evaluating always

Reply:
`ConstantExpression.List(new List())` would enter that branch and then throw

Reply:
That didn't exist when I started this branch. :p

I guess I'm not sure what constitutes a `ConstantExpression`. Its contents clearly aren't constant. Is it immutable?

Reply:
`ConstantExpression` is kind of an artifact from parsing the code. It is not constant, nor immutable. A more accurate description would be "literal". The values of these expressions are available before they need to be interpreted (generally). That's what makes them different from other expressions.

Reply:
OK, I've added a test for a constant list. But not for a constant map, because that would be mapness.